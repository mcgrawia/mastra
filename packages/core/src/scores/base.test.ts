import { describe, it, expect, beforeEach } from 'vitest';
import {
  AsyncFunctionBasedScorerBuilders,
  FunctionBasedScorerBuilders,
  MixedScorerBuilders,
  PromptBasedScorerBuilders,
} from './base.test-utils';

const createTestData = () => ({
  inputText: 'test input',
  outputText: 'test output',
  get userInput() {
    return [{ role: 'user', content: this.inputText }];
  },
  get agentOutput() {
    return { role: 'assistant', text: this.outputText };
  },
  get scoringInput() {
    return { input: this.userInput, output: this.agentOutput };
  },
});

describe('createScorer', () => {
  let testData: ReturnType<typeof createTestData>;

  beforeEach(() => {
    testData = createTestData();
  });

  describe('Steps as functions scorer', () => {
    it('should create a basic scorer with functions', async () => {
      const scorer = FunctionBasedScorerBuilders.basic;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });

    it('should create a scorer with reason', async () => {
      const scorer = FunctionBasedScorerBuilders.withReason;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });

    it('should create a scorer with preprocess and reason', async () => {
      const scorer = FunctionBasedScorerBuilders.withPreprocessAndReason;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });

    it('should create a scorer with preprocess and analyze', async () => {
      const scorer = FunctionBasedScorerBuilders.withPreprocessAndAnalyze;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });

    it('should create a scorer with preprocess only', async () => {
      const scorer = FunctionBasedScorerBuilders.withPreprocess;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });

    it('should create a scorer with preprocess, analyze, and reason', async () => {
      const scorer = FunctionBasedScorerBuilders.withPreprocessAndAnalyzeAndReason;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });

    it('should create a scorer with analyze only', async () => {
      const scorer = FunctionBasedScorerBuilders.withAnalyze;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });

    it('should create a scorer with analyze and reason', async () => {
      const scorer = FunctionBasedScorerBuilders.withAnalyzeAndReason;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });
  });

  describe('Steps as prompt objects scorer', () => {
    it('with analyze prompt object', async () => {
      const scorer = PromptBasedScorerBuilders.withAnalyze;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });

    it('with preprocess and analyze prompt object', async () => {
      const scorer = PromptBasedScorerBuilders.withPreprocessAndAnalyze;

      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });

    it('with analyze and reason prompt object', async () => {
      const scorer = PromptBasedScorerBuilders.withAnalyzeAndReason;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(typeof result.reason).toBe('string');
      expect(result).toMatchSnapshot();
    });

    it('with generate score as prompt object', async () => {
      const scorer = PromptBasedScorerBuilders.withGenerateScoreAsPromptObject;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });

    it('with all steps', async () => {
      const scorer = PromptBasedScorerBuilders.withAllSteps;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });
  });

  describe('Mixed scorer', () => {
    it('with preprocess function and analyze prompt object', async () => {
      const scorer = MixedScorerBuilders.withPreprocessFunctionAnalyzePrompt;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });

    it('with preprocess prompt and analyze function', async () => {
      const scorer = MixedScorerBuilders.withPreprocessPromptAnalyzeFunction;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });

    it('with reason function and analyze prompt', async () => {
      const scorer = MixedScorerBuilders.withReasonFunctionAnalyzePrompt;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });

    it('with reason prompt and analyze function', async () => {
      const scorer = MixedScorerBuilders.withReasonPromptAnalyzeFunction;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });
  });

  describe('Async scorer', () => {
    it('with basic', async () => {
      const scorer = AsyncFunctionBasedScorerBuilders.basic;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });

    it('with preprocess', async () => {
      const scorer = AsyncFunctionBasedScorerBuilders.withPreprocess;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });

    it('with preprocess function and analyze as prompt object', async () => {
      const scorer = AsyncFunctionBasedScorerBuilders.withPreprocessFunctionAndAnalyzePromptObject;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });

    it('with preprocess prompt object and analyze function', async () => {
      const scorer = AsyncFunctionBasedScorerBuilders.withPreprocessPromptObjectAndAnalyzeFunction;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });

    it('with async createPrompt in preprocess', async () => {
      const scorer = AsyncFunctionBasedScorerBuilders.withAsyncCreatePromptInPreprocess;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });

    it('with async createPrompt in analyze', async () => {
      const scorer = AsyncFunctionBasedScorerBuilders.withAsyncCreatePromptInAnalyze;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });

    it('with async createPrompt in generateScore', async () => {
      const scorer = AsyncFunctionBasedScorerBuilders.withAsyncCreatePromptInGenerateScore;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });

    it('with async createPrompt in generateReason', async () => {
      const scorer = AsyncFunctionBasedScorerBuilders.withAsyncCreatePromptInGenerateReason;
      const { runId, ...result } = await scorer.run(testData.scoringInput);

      expect(runId).toBeDefined();
      expect(result).toMatchSnapshot();
    });
  });
});
