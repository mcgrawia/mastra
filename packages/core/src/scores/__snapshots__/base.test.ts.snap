// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createScorer > Async scorer > with async createPrompt in analyze 1`] = `
{
  "analyzePrompt": "Async analyze prompt: TEST INPUT and TEST OUTPUT",
  "analyzeStepResult": {
    "inputFromAnalyze": "ASYNC ANALYZE INPUT",
    "outputFromAnalyze": "ASYNC ANALYZE OUTPUT",
  },
  "generateReasonPrompt": undefined,
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": undefined,
  "preprocessStepResult": {
    "reformattedInput": "TEST INPUT",
    "reformattedOutput": "TEST OUTPUT",
  },
  "reason": undefined,
  "score": 1,
}
`;

exports[`createScorer > Async scorer > with async createPrompt in generateReason 1`] = `
{
  "analyzePrompt": undefined,
  "analyzeStepResult": undefined,
  "generateReasonPrompt": "Async reason prompt: Score 1 for TEST INPUT and TEST OUTPUT",
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": undefined,
  "preprocessStepResult": {
    "reformattedInput": "TEST INPUT",
    "reformattedOutput": "TEST OUTPUT",
  },
  "reason": "This is an async reason for the score",
  "score": 1,
}
`;

exports[`createScorer > Async scorer > with async createPrompt in generateScore 1`] = `
{
  "analyzePrompt": undefined,
  "analyzeStepResult": undefined,
  "generateReasonPrompt": undefined,
  "generateScorePrompt": "Async score prompt: TEST INPUT and TEST OUTPUT",
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": undefined,
  "preprocessStepResult": {
    "reformattedInput": "TEST INPUT",
    "reformattedOutput": "TEST OUTPUT",
  },
  "reason": undefined,
  "score": 0.85,
}
`;

exports[`createScorer > Async scorer > with async createPrompt in preprocess 1`] = `
{
  "analyzePrompt": undefined,
  "analyzeStepResult": undefined,
  "generateReasonPrompt": undefined,
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": "Async prompt: test input and test output",
  "preprocessStepResult": {
    "reformattedInput": "ASYNC TEST INPUT",
    "reformattedOutput": "ASYNC TEST OUTPUT",
  },
  "reason": undefined,
  "score": 1,
}
`;

exports[`createScorer > Async scorer > with basic 1`] = `
{
  "analyzePrompt": undefined,
  "analyzeStepResult": undefined,
  "generateReasonPrompt": undefined,
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": undefined,
  "preprocessStepResult": undefined,
  "reason": undefined,
  "score": 1,
}
`;

exports[`createScorer > Async scorer > with preprocess 1`] = `
{
  "analyzePrompt": undefined,
  "analyzeStepResult": undefined,
  "generateReasonPrompt": undefined,
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": undefined,
  "preprocessStepResult": {
    "reformattedInput": "TEST INPUT",
    "reformattedOutput": "TEST OUTPUT",
  },
  "reason": undefined,
  "score": 1,
}
`;

exports[`createScorer > Async scorer > with preprocess function and analyze as prompt object 1`] = `
{
  "analyzePrompt": "Analyze the input and output: test input and test output",
  "analyzeStepResult": {
    "inputFromAnalyze": "TEST INPUT",
    "outputFromAnalyze": "TEST OUTPUT",
  },
  "generateReasonPrompt": undefined,
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": undefined,
  "preprocessStepResult": {
    "reformattedInput": "TEST INPUT",
    "reformattedOutput": "TEST OUTPUT",
  },
  "reason": undefined,
  "score": 1,
}
`;

exports[`createScorer > Async scorer > with preprocess prompt object and analyze function 1`] = `
{
  "analyzePrompt": undefined,
  "analyzeStepResult": {
    "inputFromAnalyze": "TEST INPUT",
    "outputFromAnalyze": "TEST OUTPUT",
  },
  "generateReasonPrompt": undefined,
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": "Analyze the input and output: test input and test output",
  "preprocessStepResult": {
    "reformattedInput": "TEST INPUT",
    "reformattedOutput": "TEST OUTPUT",
  },
  "reason": undefined,
  "score": 1,
}
`;

exports[`createScorer > Mixed scorer > with preprocess function and analyze prompt object 1`] = `
{
  "analyzePrompt": "Test Analyze prompt",
  "analyzeStepResult": {
    "inputLength": 10,
    "outputLength": 11,
  },
  "generateReasonPrompt": undefined,
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": undefined,
  "preprocessStepResult": {
    "reformattedInput": "TEST INPUT from preprocess function!",
    "reformattedOutput": "TEST OUTPUT from preprocess function!",
  },
  "reason": undefined,
  "score": 1,
}
`;

exports[`createScorer > Mixed scorer > with preprocess prompt and analyze function 1`] = `
{
  "analyzePrompt": undefined,
  "analyzeStepResult": {
    "inputFromAnalyze": "TEST INPUT from preprocess prompt!!",
    "outputFromAnalyze": "TEST OUTPUT from preprocess prompt!!",
  },
  "generateReasonPrompt": undefined,
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": "Test Preprocess prompt",
  "preprocessStepResult": {
    "reformattedInput": "TEST INPUT from preprocess prompt!",
    "reformattedOutput": "TEST OUTPUT from preprocess prompt!",
  },
  "reason": undefined,
  "score": 1,
}
`;

exports[`createScorer > Mixed scorer > with reason function and analyze prompt 1`] = `
{
  "analyzePrompt": "Test Analyze prompt",
  "analyzeStepResult": {
    "inputLength": 10,
    "outputLength": 11,
  },
  "generateReasonPrompt": undefined,
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": undefined,
  "preprocessStepResult": undefined,
  "reason": "the reason is because the input is 10 and the output is 11 from generateReason function",
  "score": 1,
}
`;

exports[`createScorer > Mixed scorer > with reason prompt and analyze function 1`] = `
{
  "analyzePrompt": undefined,
  "analyzeStepResult": {
    "inputFromAnalyze": "test input from analyze function!",
    "outputFromAnalyze": "test output from analyze function!",
  },
  "generateReasonPrompt": "Test Generate Reason prompt",
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": undefined,
  "preprocessStepResult": undefined,
  "reason": "This is the reason.",
  "score": 1,
}
`;

exports[`createScorer > Steps as functions scorer > should create a basic scorer with functions 1`] = `
{
  "analyzePrompt": undefined,
  "analyzeStepResult": undefined,
  "generateReasonPrompt": undefined,
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": undefined,
  "preprocessStepResult": undefined,
  "reason": undefined,
  "score": 1,
}
`;

exports[`createScorer > Steps as functions scorer > should create a scorer with analyze and reason 1`] = `
{
  "analyzePrompt": undefined,
  "analyzeStepResult": {
    "inputFromAnalyze": "test input!",
    "outputFromAnalyze": "test output!",
  },
  "generateReasonPrompt": undefined,
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": undefined,
  "preprocessStepResult": undefined,
  "reason": "the reason the score is 1 is because the input is test input! and the output is test output!",
  "score": 1,
}
`;

exports[`createScorer > Steps as functions scorer > should create a scorer with analyze only 1`] = `
{
  "analyzePrompt": undefined,
  "analyzeStepResult": {
    "inputFromAnalyze": "test input!",
    "outputFromAnalyze": "test output!",
  },
  "generateReasonPrompt": undefined,
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": undefined,
  "preprocessStepResult": undefined,
  "reason": undefined,
  "score": 1,
}
`;

exports[`createScorer > Steps as functions scorer > should create a scorer with preprocess and analyze 1`] = `
{
  "analyzePrompt": undefined,
  "analyzeStepResult": {
    "inputFromAnalyze": "TEST INPUT!",
    "outputFromAnalyze": "TEST OUTPUT!",
  },
  "generateReasonPrompt": undefined,
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": undefined,
  "preprocessStepResult": {
    "reformattedInput": "TEST INPUT",
    "reformattedOutput": "TEST OUTPUT",
  },
  "reason": undefined,
  "score": 1,
}
`;

exports[`createScorer > Steps as functions scorer > should create a scorer with preprocess and reason 1`] = `
{
  "analyzePrompt": undefined,
  "analyzeStepResult": undefined,
  "generateReasonPrompt": undefined,
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": undefined,
  "preprocessStepResult": {
    "reformattedInput": "TEST INPUT",
    "reformattedOutput": "TEST OUTPUT",
  },
  "reason": "the reason the score is 1 is because the input is TEST INPUT and the output is TEST OUTPUT",
  "score": 1,
}
`;

exports[`createScorer > Steps as functions scorer > should create a scorer with preprocess only 1`] = `
{
  "analyzePrompt": undefined,
  "analyzeStepResult": undefined,
  "generateReasonPrompt": undefined,
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": undefined,
  "preprocessStepResult": {
    "reformattedInput": "TEST INPUT",
    "reformattedOutput": "TEST OUTPUT",
  },
  "reason": undefined,
  "score": 1,
}
`;

exports[`createScorer > Steps as functions scorer > should create a scorer with preprocess, analyze, and reason 1`] = `
{
  "analyzePrompt": undefined,
  "analyzeStepResult": {
    "inputFromAnalyze": "TEST INPUT!",
    "outputFromAnalyze": "TEST OUTPUT!",
  },
  "generateReasonPrompt": undefined,
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": undefined,
  "preprocessStepResult": {
    "reformattedInput": "TEST INPUT",
    "reformattedOutput": "TEST OUTPUT",
  },
  "reason": "the reason the score is 1 is because the input is TEST INPUT! and the output is TEST OUTPUT!",
  "score": 1,
}
`;

exports[`createScorer > Steps as functions scorer > should create a scorer with reason 1`] = `
{
  "analyzePrompt": undefined,
  "analyzeStepResult": undefined,
  "generateReasonPrompt": undefined,
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": undefined,
  "preprocessStepResult": undefined,
  "reason": "the reason the score is 1 is because the input is test input and the output is test output",
  "score": 1,
}
`;

exports[`createScorer > Steps as prompt objects scorer > with all steps 1`] = `
{
  "analyzePrompt": "Test Analyze prompt",
  "analyzeStepResult": {
    "inputLength": 10,
    "outputLength": 11,
  },
  "generateReasonPrompt": "Test Generate Reason prompt",
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": "Test Preprocess prompt",
  "preprocessStepResult": {
    "reformattedInput": "TEST INPUT",
    "reformattedOutput": "TEST OUTPUT",
  },
  "reason": "this is a reason.",
  "score": 1,
}
`;

exports[`createScorer > Steps as prompt objects scorer > with analyze and reason prompt object 1`] = `
{
  "analyzePrompt": "Test Analyze prompt",
  "analyzeStepResult": {
    "inputLength": 10,
    "outputLength": 11,
  },
  "generateReasonPrompt": "Test Generate Reason prompt",
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": undefined,
  "preprocessStepResult": undefined,
  "reason": "This is a test reason",
  "score": 1,
}
`;

exports[`createScorer > Steps as prompt objects scorer > with analyze prompt object 1`] = `
{
  "analyzePrompt": "Test Analyze prompt",
  "analyzeStepResult": {
    "inputLength": 10,
    "outputLength": 11,
  },
  "generateReasonPrompt": undefined,
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": undefined,
  "preprocessStepResult": undefined,
  "reason": undefined,
  "score": 1,
}
`;

exports[`createScorer > Steps as prompt objects scorer > with generate score as prompt object 1`] = `
{
  "analyzePrompt": undefined,
  "analyzeStepResult": undefined,
  "generateReasonPrompt": undefined,
  "generateScorePrompt": "Test Generate Score prompt",
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": undefined,
  "preprocessStepResult": undefined,
  "reason": "the reason the score is 1 is because the input is [{"role":"user","content":"test input"}] and the output is {"role":"assistant","text":"test output"}",
  "score": 1,
}
`;

exports[`createScorer > Steps as prompt objects scorer > with preprocess and analyze prompt object 1`] = `
{
  "analyzePrompt": "Test Analyze prompt",
  "analyzeStepResult": {
    "inputLength": 10,
    "outputLength": 11,
  },
  "generateReasonPrompt": undefined,
  "generateScorePrompt": undefined,
  "input": [
    {
      "content": "test input",
      "role": "user",
    },
  ],
  "output": {
    "role": "assistant",
    "text": "test output",
  },
  "preprocessPrompt": "Test Preprocess prompt",
  "preprocessStepResult": {
    "reformattedInput": "TEST INPUT",
    "reformattedOutput": "TEST OUTPUT",
  },
  "reason": undefined,
  "score": 1,
}
`;
